StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/SpacingRules/SA1023CSharp9UnitTests.cs

```diff
@@ -3,9 +3,74 @@
 
 namespace StyleCop.Analyzers.Test.CSharp9.SpacingRules
 {
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis.Testing;
     using StyleCop.Analyzers.Test.CSharp8.SpacingRules;
+    using Xunit;
+    using static StyleCop.Analyzers.SpacingRules.SA1023DereferenceAndAccessOfSymbolsMustBeSpacedCorrectly;
+    using static StyleCop.Analyzers.Test.Verifiers.StyleCopCodeFixVerifier<
+        StyleCop.Analyzers.SpacingRules.SA1023DereferenceAndAccessOfSymbolsMustBeSpacedCorrectly,
+        StyleCop.Analyzers.SpacingRules.TokenSpacingCodeFixProvider>;
 
     public class SA1023CSharp9UnitTests : SA1023CSharp8UnitTests
     {
+        [Fact]
+        public async Task TestFunctionPointerParameterInvalidSpacingAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    unsafe delegate*<int {|#0:*|}> FuncPtr1;
+    unsafe delegate*<int{|#1:*|} > FuncPtr2;
+}
+";
+
+            var fixedCode = @"public class TestClass
+{
+    unsafe delegate*<int*> FuncPtr1;
+    unsafe delegate*<int*> FuncPtr2;
+}
+";
+
+            var expected = new[]
+            {
+                Diagnostic(DescriptorNotPreceded).WithLocation(0),
+                Diagnostic(DescriptorNotFollowed).WithLocation(1),
+            };
+
+            await VerifyCSharpFixAsync(testCode, expected, fixedCode, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestFunctionPointerTypeInvalidSpacingAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    unsafe delegate {|#0:*|}<int*> FuncPtr1;
+    unsafe delegate{|#1:*|} <int*> FuncPtr2;
+    unsafe delegate {|#2:*|} managed<int*> FuncPtr3;
+    unsafe delegate{|#3:*|}managed<int*> FuncPtr4;
```

Comment:
💡 IIRC, the `unmanaged` keyword only works for .NET 5 and newer. I believe there are some instances of tests in the repository using `ReferenceAssemblies.Net.Net50` explicitly. Considering function pointers are only officially supported for .NET 5 and newer, it would be fine to just update this test to always use it and test both `managed` and `unmanaged` cases. I'm happy to leave this for a follow-up PR though.

Reply:
I'm not sure I get your point. C# 9.0 is supported only on .NET 5 and newer. Reference assemblies are not needed in this case since the function pointer is not a new invention at the IL level.

Reply:
Well... error CS8889: The target runtime doesn't support extensible or runtime-environment default calling conventions.

Reply:
It's looking for this field, which is only available in .NET 5 and newer:
https://github.com/dotnet/runtime/blob/2445324a501efaca19e136b7a7ca46d8ab5d95ec/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeFeature.cs#L18-L21

Reply:
Shall I specify a calling convention to make the build pass as in #3254?

Reply:
I'd prefer the test be updated to this form instead:

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/6399f4986f460445e1d12054cedfb62d4c9c605a/StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/NamingRules/SA1300CSharp9UnitTests.cs#L41-L54